doc/develop/coding-guidelines.rst

```diff
@@ -0,0 +1,213 @@
+the wxPython style, which follows the wxwidget style from
+typical C++ conventions.
+
+
+For e.g. 1.1 the VAs in odemis have the following structure
+
+.. code-block:: python
+
+ self.pixelSize = model.VigilantAttribute(pxs, unit="m", readonly=True)
+ self.horizontalFoV = model.FloatContinuous(hfv, range=[10e-9, 10e-3],
+                                               unit="m")
+ self.magnification = model.VigilantAttribute(self._hfw_nomag / hfv,
+                                                 unit="", readonly=True)
+
+* Naming of test files follow the below ways to provide consistency for files in test folders (no _init_.py file), which are included in each sub-directories of src folder.
+
+ * A unittest convention is used to name the test class for a module as TestModuleName(unittest.Testcase).
+
+ * Class methods are named as test_method_name().
+
+ * In a base class test which is used for repeating test cases with different input settings, we name the base class as TestModuleName.
+
+For e.g. 1.2 the naming of a base class test is as follows
+
+.. code-block:: python
+
+    class LakeshoreBaseTest:
+        """Base class for testing different models of LakeShore temperature controller."""
+
+        @classmethod
+        def tearDownClass(cls):
+            cls.dev.terminate()  # free up socket.
+
+In class, mark the attributes as private with a first letter as _. They are
+accessible normally from other classes, but by convention, as developers we know that they are
+private to the classes and should not be externally modified.
+If there are public attributes, the attributes might have to be changed in several
+files and folders.
+
+The usage of public attributes is useful when their access is required by a third
+party or another class. In this scenario, public attributes can be used.
+
+For e.g. 1.2 the implementation of public and private attributes
+
+.. code-block:: python
+
+    class Pokemon:
+        """
+        Links special powers to the given pokemon type
+
+        """
+       def __init__(self, name, alias):
+          self.name = name       # public
+          self._alias = alias   # private nomenclature
+
+       def print_who(self):
+          print('name  : ', self.name)
+          print('alias : ', self._alias)
+
+
+Docstrings
+==================
+
+Use three double quotes for the docstring. The docstring consists of function definition, description of input and return arguments. We will follow the reStructuredText style to write the docstring.
+For e.g. 2.1 when type is declared in the docstring
+
+.. code-block:: python
+
+    def get_equality(arg1, arg2):
+        """Summary line.
+
+        Extended description of function.
+
+        :param arg1: (int) Description of arg1.
+        :param arg2: (str) Description of arg2.
+        :raise: ValueError if arg1 is equal to arg2
+        :return: (bool) Description of return value
+
+        """
+
```

Comment:
Do we want an empty line after the docstring? I would tend to say "no". So let's drop that line.